crates/ruma-common/src/events/room/message.rs

```diff
@@ -230,8 +221,11 @@ impl RoomMessageEventContent {
             _ => message,
         };
 
-        let relates_to = if let Some(Relation::Thread(Thread { event_id, .. })) =
-            original_message.content.relates_to.as_ref().filter(|_| forward_thread)
+        let relates_to = if let Some(Relation::Thread(Thread { event_id, .. })) = original_message
+            .content
+            .relates_to
+            .as_ref()
+            .filter(|_| forward_thread == ForwardThread::Yes)
```

Comment:
I hadn't noticed this ugly hack around lack of let-chains. It really can't wait for that to be stabilized!